packages/openapi-generator/src/generator.ts

```diff
@@ -275,26 +245,44 @@ async function generateService(
   logger.info(`Successfully generated client for '${inputFilePath}'`);
 }
 
+function getAbsoluteAndRelativePath(
+  absolutePath: string
+): AbsoluteAndRelativePath {
+  return { absolutePath, relativePath: relative(process.cwd(), absolutePath) };
+}
+
 /**
  * Recursively searches through a given input path and returns all file paths as a string array.
  * @param input the path to the input directory.
  * @returns all file paths as a string array.
  */
-export async function getInputFilePaths(input: string): Promise<string[]> {
+export async function getInputFilePaths(
+  input: string
+): Promise<AbsoluteAndRelativePath[]> {
   if ((await lstat(input)).isFile()) {
-    return [input];
+    return [getAbsoluteAndRelativePath(input)];
   }
 
-  const directoryContents = await readdir(input);
+  const absoluteInputPaths = await readdir(input);
+  const directoryContents: AbsoluteAndRelativePath[] = absoluteInputPaths.map(
+    path => getAbsoluteAndRelativePath(path)
+  );
   return directoryContents.reduce(
-    async (paths: Promise<string[]>, directoryContent) => [
+    async (paths: Promise<AbsoluteAndRelativePath[]>, directoryContent) => [
       ...(await paths),
-      ...(await getInputFilePaths(resolve(input, directoryContent)))
+      ...(await getInputFilePaths(
+        resolve(input, directoryContent.absolutePath)
+      ))
     ],
-    Promise.resolve([])
+    Promise.resolve([] as AbsoluteAndRelativePath[])
   );
 }
 
+export interface AbsoluteAndRelativePath {
+  relativePath: string;
+  absolutePath: string;
+}
```

Comment:
I am not a fan of this. Seems more complex than before. Why did you decide to go this way?

Reply:
hm I did not want to calculate the relative path multiple times.